### ISSUE TITLE
Can there be way some standard way to mark an annotation as only usable in an annotation?

### ISSUE THREAD

**Issue by parlough** [dart-lang/sdk#54691]
As otherwise they get suggested in areas where they don't make sense. Particularly egregious is `pragma` which comes up first in a statement context typing `pr^` but you want to print something out.

This functionality would enable the analysis server to only suggest them in a metadata context (`@^`) or a constant assignment context (`const newAnnotation = ^`).

**Comment by srawlins** [comment]
I've never thought of this before, but I love it. Something like `@annotation`. CC @bwilkerson

**Comment by parlough** [comment]
Yeah something like that seems great to me! The one problem is that many of these that Dart developers interact with, including my example above (`pragma`), are defined in the SDK. So it'd be preferred to have a solution that works there as well.

**Comment by lrhn** [comment]
`@pragma("annotation")` 😁
